### PR TITLE
adding basic code coverage to project

### PR DESCRIPTION
**Pull request by biswassri**
[MGDSTRM -6791](https://issues.redhat.com/browse/MGDSTRM-6791)- Adding code coverage as a replication of how it is setup in kas-fleetshard.

### CODE REVIEW

File: systemtests/pom.xml [hunk 1]
```diff
@@ -14,6 +14,7 @@
         <!-- System test image dependencies -->
         <keycloak.image>quay.io/keycloak/keycloak:14.0.0</keycloak.image>
         <strimzi-kafka.tag>quay.io/strimzi/kafka:0.26.0-kafka-3.0.0</strimzi-kafka.tag>
+        <jacoco.skip>true</jacoco.skip>
     </properties>
```
Comment (MikeEdgar) [inline systemtests/pom.xml:18]: I don't think we should skip Jacoco completely, but rather skip the classes/packages in the systemtests project. We'll want to gather the coverage of the `kafka-admin` module for the tests executed in systemtests.

https://www.baeldung.com/jacoco-report-exclude

Comment (biswassri) [reply systemtests/pom.xml:18]: So, when using the exclude parameter, the rule fails for the systemtests. It returns a 0.0 for the line coverage.

Comment (biswassri) [reply systemtests/pom.xml]: > rather skip the classes/packages in the systemtests project.

I couldn't exclude it from `<rule>` but I was able to skip the above.

**Comment by MikeEdgar** [comment]
> I am noticing some warnings for the classes generated by kafka-admin `model` package. Should that be a concern

I think it's ok for now. The warnings are because the coverage is collected from instrumented versions of the class files and the report uses the originals. From the output they are just the model classes, so let's ignore it for now.

**Comment by MikeEdgar** [comment]
@biswassri I looked at this locally a bit yesterday and I think it needs to be a bit more complex to fully capture the coverage. We want to run the `check` goal for coverage in the `kafka-admin` project for the aggregate of the tests run in both `kafka-admin` and `systemtests`. 
To do this, I think we need to have `kafka-admin` run the `prepare-agent` and `prepare-agent-integration` goals only. Then the `systemtests` project should run `prepare-agent`, `prepare-agent-integration`, `report-aggregate` (phase=verify), then `check`. I believe we will also need to run the `merge` goal before `check`, but I'm not certain of that.

**Comment by biswassri** [comment]
Replacing this PR with a new PR to test out SonarCloud.

**Pull request closed without merging**